### PR TITLE
Image set index

### PR DESCRIPTION
**Pull request by iezer**
@bantic  For you CC @redyaffle 

DO NOT MERGE. This will need a rebase after merging the google maps branch. But it is tested and ready for a review.

Main feature is an Image Set Index page which is more in line with the specs. Has a search field and a simple list of image sets with an active image set and actions at the top. I created a reusable `lg-selectable` component for this type of thing, and refactored the `lg-lion-search` component so it can be pretty much re-used to search image-sets as well. @redyaffle  Would appreciate your help sprucing up the style here at some point.

Also I added some logic to the image-set page to show the lion name if available and only show the is-verified flag if there is a lon.

Here's what it looks like. This requires the `image-set-search` API branch which has a separate [open PR](https://github.com/201-created/LG-api/pull/10)

### CODE REVIEW

File: app/components/lg-lion-search.js [hunk 1]
```diff
@@ -58,8 +60,14 @@ export default Ember.Component.extend({
...
-      var params = this.get('params');
-      this.sendAction('action', params);
+      var params = this.get('params'),
+          store = this.get('store'),
+          component = this,
+          searchModel = this.get('searchModel');
+
+      store.find(searchModel, params).then(function(results){
+        component.sendAction('action', results);
+      });
```
Comment (bantic) [inline app/components/lg-lion-search.js]: we'll definitely need to handle errors, too. A generic "there was an error message" is probably sufficient.

File: app/controllers/image-sets/index.js [hunk 2]
```diff
@@ -1,6 +1,57 @@
...
+      var activeImageSet = this.get('activeImageSet');
+      if (activeImageSet) {
+        this.transitionToRoute('image-set.cv-results', activeImageSet);
+      }
+    },
+
+    requestCv: function(activeImageSet) {
+      if (activeImageSet) {
+        // Bubble up to handler on application route
+        return true;
```
Comment (bantic) [inline app/controllers/image-sets/index.js:53]: nice, and good comment

File: app/routes/image-sets/index.js [hunk 3]
```diff
@@ -10,9 +10,12 @@ export default Ember.Route.extend({
 
   setupController: function(controller, model) {
     var currentUser = this.get('toriiSession.currentUser');
-    controller.setProperties({
-      currentUser: currentUser,
-      model: model
+    return this.store.find('organization').then(function(organizations){
```
Comment (bantic) [inline app/routes/image-sets/index.js]: is there a reason this isn't done in the model or beforemodel hook instead?

File: app/templates/image-sets/index.hbs [hunk 4]
```diff
@@ -1,6 +1,52 @@
 <h2>Image Sets for {{currentOrganization.name}}</h2>
-{{#if model}}
-  {{#each model as |imageSet|}}
-    {{lg-image-set-summary imageSet=imageSet}}
-  {{/each}}
-{{/if}}
+
+{{lg-lion-search action='displayResults' searchModel='imageSet' organizations=organizations store=store selectedOrganization=selectedOrganization}}
```
Comment (bantic) [inline app/templates/image-sets/index.hbs:3]: yeah, it's great to have this as a self-contained component. nice work

File: tests/acceptance/image-sets-test.js [hunk 5]
```diff
@@ -47,21 +57,94 @@ test('delete /image-set/:id', function() {
...
+test('test request CV', function() {
+  expect(6);
+
+  stubRequest('post', '/cvRequests', function(request){
+    ok(true, 'cv request post api was called');
+    return this.noContent(201);
+  });
+
+  signIn();
+  visit('/image-sets');
```
Comment (bantic) [inline tests/acceptance/image-sets-test.js]: might be worth making a `signInAndVisit(url)` async helper...

**Comment by bantic** [comment]
:+1: very nice. 

giphy

**Comment by iezer** [comment]
@bantic  Addressed your comments, added a signInAndVisit helper and alert for search errors. Merging. Thanks for your feedback.

**Pull request merged**